### PR TITLE
Sonarcloud issues fixes

### PR DESCRIPTION
**Pull request by MichalKinas**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

Fixes most of issues reported in https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=Alfresco_alfresco-content-app

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by nikita-web-ua — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

File: e2e/protractor/suites/actions/copy-move/copy.test.ts [hunk 1]
```diff
@@ -246,42 +246,72 @@ describe('Copy content', () => {
...
   });
 
-  async function copyFile(fileName: string, location: string = '', destination: string, doBefore?: () => void) {
-    if (doBefore) {
-      await doBefore();
+  async function baseCopy(
+    itemName: string | string[],
+    location: string,
+    destination: string,
+    undo: boolean = false,
```
Comment (AleksanderSklorz) [inline e2e/protractor/suites/actions/copy-move/copy.test.ts:253]: no need for type when you specify default value

File: e2e/protractor/suites/actions/copy-move/copy.test.ts [hunk 2]
```diff
@@ -246,42 +246,72 @@ describe('Copy content', () => {
...
+    expectedMsg: string = 'Copied 1 item'
```
Comment (AleksanderSklorz) [inline e2e/protractor/suites/actions/copy-move/copy.test.ts:254]: same here - no need for type when default value is used

File: e2e/protractor/suites/actions/copy-move/copy.test.ts [hunk 3]
```diff
@@ -246,42 +246,72 @@ describe('Copy content', () => {
...
+    undo?: boolean,
+    expectedMsg?: string
+  ) {
+    if (doBefore) {
+      await doBefore();
+    }
+    await baseCopy(itemName, location, destination, undo, expectedMsg);
+  }
+
+  async function copyFile(fileName: string, location: string = '', destination: string, doBefore?: () => void) {
```
Comment (AleksanderSklorz) [inline e2e/protractor/suites/actions/copy-move/copy.test.ts:305]: can we remove type from location as default value is applied?

File: e2e/protractor/suites/actions/copy-move/copy.test.ts (hunk 1 above)
Comment (MichalKinas) [reply e2e/protractor/suites/actions/copy-move/copy.test.ts]: done

File: e2e/protractor/suites/actions/copy-move/copy.test.ts (hunk 2 above)
Comment (MichalKinas) [reply e2e/protractor/suites/actions/copy-move/copy.test.ts]: done

File: e2e/protractor/suites/actions/copy-move/copy.test.ts (hunk 3 above)
Comment (MichalKinas) [reply e2e/protractor/suites/actions/copy-move/copy.test.ts]: done

**Review by DenysVuika — APPROVED** [review]

**Review by nikita-web-ua — APPROVED** [review]

File: projects/aca-content/src/lib/components/common/toggle-shared/toggle-shared.component.ts [hunk 4]
```diff
@@ -62,7 +62,7 @@ export class ToggleSharedComponent implements OnInit, OnDestroy {
       this.selectionState = selectionState;
 
       this.isShared =
-        (this.selectionState.first && this.selectionState.first.entry && (this.selectionState.first.entry as any).sharedByUser) ||
+        (this.selectionState?.first?.entry && (this.selectionState.first.entry as any).sharedByUser) ||
```
Comment (AleksanderSklorz) [inline projects/aca-content/src/lib/components/common/toggle-shared/toggle-shared.component.ts:65]: this line can be reduced to just (this.selectionState?.first?.entry as any)?.sharedByUser
(no need to repeat this.selectionState.first.entry twice)

**Review by AleksanderSklorz — APPROVED** [review]

**Comment by DenysVuika** [comment]
One e2e constantly failing, probably affected by the changes

**Pull request merged**